### PR TITLE
Upgrade vega tooltips to use custom tooltip format

### LINKED ISSUE #254: [FEATURE] Upgrade Vega-tooltips plugin version
**Is your feature request related to a problem?**
The current version of the vega-tooltip package is 0.24.2 and doesn't support customizing the vega-tooltip.

**What solution would you like?**
Update the vega-tooltip version to 0.30.0 which does support customizing via formatTooltip API
https://github.com/vega/vega-tooltip/blob/next/docs/APIs.md#options

### PR DESCRIPTION
**Pull request by jovancacvetkovic**
### Description
Upgrades vega tooltips to use custom tooltip format

### Issues Resolved
Closes #254

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jovancacvetkovic** [comment]
@amsiglan note that this is only for code preview. The vega tooltip needs an upgrade in order for this to work.

File: public/utils/helpers.tsx [hunk 1]
```diff
@@ -165,14 +166,65 @@ export function renderVisualization(spec: TopLevelSpec, containerId: string) {
   }
 
   function renderVegaSpec(spec: {}) {
-    const handler = new Handler();
+    let pathGroup: any[] = [];
```
Comment (amsiglan) [inline public/utils/helpers.tsx:169]: Doubt: Is `pathGroup` a term specific to Vega tooltip? If not can we name it something that describes what it means?

File: public/utils/helpers.tsx [hunk 2]
```diff
@@ -165,14 +166,65 @@ export function renderVisualization(spec: TopLevelSpec, containerId: string) {
...
+        let rowData = '';
+        values.forEach((item: any) => {
+          rowData += `
+            <tr>
+              <td>${sanitize(item[0])}</td>
+              <td>${sanitize(item[1])}</td>
+            </tr>
+          `;
+        });
+        let tableData = `<table>${rowData}</table>`;
```
Comment (amsiglan) [inline public/utils/helpers.tsx:193]: make this a `const`?

File: public/utils/helpers.tsx (hunk 1 above)
Comment (jovancacvetkovic) [reply public/utils/helpers.tsx:169]: It is specific to charts in general, as charts consist of a bunch of paths, and here I filter only the paths that contain the colors to update the tooltip. The path is a general name for chart elements.
Something more intuitive perhaps: chartColoredItems or chartColors or tooltipColors?

File: public/utils/helpers.tsx (hunk 2 above)
Comment (jovancacvetkovic) [reply public/utils/helpers.tsx]: Done.

File: public/utils/helpers.tsx (hunk 1 above)
Comment (jovancacvetkovic) [reply public/utils/helpers.tsx:170]: @amsiglan any thoughts?

**Comment by jovancacvetkovic** [comment]
@AWSHurneyt this PR has to wait for [PR #3359](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3359)

**Comment by jovancacvetkovic** [comment]
@amsiglan this [PR #3359](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3359) is merged.

File: public/utils/helpers.tsx (hunk 1 above)
Comment (amsiglan) [reply public/utils/helpers.tsx:170]: `chartColoredItems` seems apt here

Comment (jovancacvetkovic) [reply public/utils/helpers.tsx]: Updated.

**Comment by jovancacvetkovic** [comment]
@amsiglan PR has been updated and tested with the latest Dashboard main branch.

**Review by amsiglan — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**